modules/nlu/src/backend/application/training-queue.ts

```diff
@@ -0,0 +1,287 @@
+import * as sdk from 'botpress/sdk'
+import * as NLU from 'common/nlu/engine'
+import _ from 'lodash'
+
+import moment from 'moment'
+import ms from 'ms'
+import nanoid from 'nanoid'
+import { ITrainingRepository, ITransactionContext } from './training-repo'
+import { TrainingId, TrainerService, TrainingListener, TrainingState, TrainingSession, I } from './typings'
+
+export interface TrainingQueueOptions {
+  maxTraining: number
+}
+
+const DEFAULT_OPTIONS: TrainingQueueOptions = {
+  maxTraining: 2
+}
+
+const DEFAULT_STATE = { status: <sdk.NLU.TrainingStatus>'idle', progress: 0 }
+
+export type ITrainingQueue = I<TrainingQueue>
+
+const debug = DEBUG('nlu').sub('lifecycle')
+const TRAINING_LIFE_TIME = ms('5m')
```

Comment:
really ? sadly I've seen longer training sessions .... will that crash ?

Reply:
this is badly named, it should represent the max time between progress calls